src/azul/service/index_query_service.py

```diff
@@ -74,10 +76,35 @@ def get_data(self,
                     file['url'] = file_url_func(catalog=catalog,
                                                 file_uuid=file['uuid'],
                                                 version=file['version'])
+        if entity_type == 'projects':
+            for hit in response['hits']:
+                # Compose a URL for each file in stratification trees
+                for project in hit['projects']:
+                    self.transform_stratification_file_nodes(tree=project['contributorMatrices'],
```

Comment:
This is ok but we should try to make this even more generic. I'm thinking of something that can scan the entire response for file references and inject a URL to the file. A file reference could be a dictionary containing both `file_uuid` and `file_version` key. A recursive traversal would solve this new use case and the preexisting one right above this section. The only issue is that the file references for the preexisting use case use the `uuid` and `version` keys instead of `file_uuid` and `file_version` but I'm sure that can be fixed.